File: Huawei/拼接URL.java

/**最大乘积
 * @author ASUS*/

/**
 * 拼接字符串
 */

public class Main {
    public static void main(String[] args) {
//        String input = ",/";
        String input = "/acm,/bb";
        String wordArr[] = input.split(",");

        if (wordArr.length == 0) {
            System.out.println("/");
            return;
        }

        if (wordArr.length == 1) {
            System.out.println("/" + wordArr[0]);
            return;
        }
        String prefix = wordArr[0];
        if (prefix.startsWith("/"))
            prefix = prefix.replaceFirst("/", "");
        String suffix = wordArr[1];
        if (suffix.startsWith("/"))
            suffix = suffix.replaceFirst("/", "");
        System.out.println("/" + prefix + "/" + suffix);
    }
}

File: Huawei/拼接URL.py


prefix, suffix = input().split(",")
print(prefix, suffix)


def result():
    global prefix # 声明前缀
    global suffix # 声明后缀
    if len(prefix) == 0: # 前缀没有
        prefix = "/"
    if len(suffix) == 0: # 后缀没有
        suffix = "/"
    if prefix[-1] == "/" and suffix[0] == "/": # 如果前缀结尾和后缀开头都为/，需要自动去重
        return prefix + suffix[1:]
    if prefix[-1] != "/" and suffix[0] != "/": # 如果前缀结尾和后缀开头都没有/，需要自动补上/连接符
        return prefix + "/" + suffix
    # 前缀末尾有"/",后缀开头无"/"
    # 前缀末尾无"/",后缀开头有"/"
    return prefix+suffix
print(result())